### PR TITLE
Clean a few things in the plugin manager

### PR DESCRIPTION
**Pull request by Myoldmopar**
Pull request overview
---------------------
Unit tests were failing when run serially because a function pointer was not reset to null.  Cleaned up a few other things along the way.

### CODE REVIEW

File: src/EnergyPlus/PluginManager.cc [hunk 1]
```diff
@@ -724,9 +724,41 @@ namespace PluginManagement {
         globalVariableNames.clear();
         globalVariableValues.clear();
         plugins.clear();
-        pluginManager.reset(); // delete the current plugin manager instance, which was created in simulation manager, this clean up Python
         PluginManagement::fullyReady = false;
         PluginManagement::apiErrorFlag = false;
+        PluginManager * p = PluginManagement::pluginManager.release();
```
Comment (Myoldmopar) [inline src/EnergyPlus/PluginManager.cc:729]: Release the unique pointer ownership and then call delete on the pointer itself.

File: src/EnergyPlus/PluginManager.cc [hunk 2]
```diff
@@ -724,9 +724,41 @@ namespace PluginManagement {
...
+        delete p;
+        wrapperDLLHandle = nullptr;
```
Comment (Myoldmopar) [inline src/EnergyPlus/PluginManager.cc:731]: Reset all the DLL and function pointers to null.

File: src/EnergyPlus/PluginManager.cc [hunk 3]
```diff
@@ -929,11 +961,11 @@ namespace PluginManagement {
     PluginManager::~PluginManager()
     {
 #if LINK_WITH_PYTHON
-        //(*EP_Py_FinalizeEx)();
+        if (EP_Py_FinalizeEx) (*EP_Py_FinalizeEx)();
```
Comment (Myoldmopar) [inline src/EnergyPlus/PluginManager.cc:964]: Only call these functions if the DLL has been opened.

File: src/EnergyPlus/PluginManager.hh [hunk 4]
```diff
@@ -196,38 +196,6 @@ namespace PluginManagement {
     extern bool fullyReady;
     extern bool apiErrorFlag;
 
-    // lazy pointers into the Python DLL
```
Comment (Myoldmopar) [inline src/EnergyPlus/PluginManager.hh:199]: These never needed to be extern-d in the header.

**Review by mjwitte — APPROVED** [review]
I don't pretend to understand all of the changes here, but it fixes the problem - all unit tests run locally. Not waiting for anymore CI reports.

**Pull request merged**